### ISSUE TITLE
A more customizable and interactive TensorBoard 

### ISSUE THREAD

**Issue by dclong** [tensorflow/tensorboard#5333]
When a user log information to TensorBoard for visualization, instead of justing logging the data, the user is often force to think or make decision on how to visualize  data. For example, should I combine training loss and validation loss into 1 plot or keep them as separate plots? What "tag" (or title) should I give to plots. IMHO, this is a coupling behavior. When a user develop his/her training code, he/she should only worry about "what" kind of information he want to log but not how to visualize it. The "how" to visualize part should be take care by TensorBoard. 

To enable this decoupling design, TensorBoard need to support more customizable and interactive ways to users to create or editing visualizations. For example, while a model is being trained, a user can leverage TensorBoard to explore training/validation loss by importing both of them into a plot (since they have identical x-axis). If the magnitude of the training/validation losses are too big and makes it impossible to see the trend of one of them, then the user can just create 2 separate plots (1 for training loss and 1 for validation loss). The similar concept can be extended to many existing visualizations. 

As a matter of fact, there are already existing requests/discussions related to this one. 

https://github.com/tensorflow/tensorboard/issues/893

https://stackoverflow.com/questions/60079644/how-do-you-edit-an-existing-tensorboard-training-loss-summary

https://stackoverflow.com/questions/58276718/how-to-load-selected-range-of-samples-in-tensorboard

**Comment by dclong** [comment]
And it would be great if TensorBoard allows users to click buttons to upload customized images, etc.
